secant/AppDelegate.swift

```diff
@@ -0,0 +1,189 @@
+        ).logging()
+    }
+
+    func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil
+    ) -> Bool {
+#if DEBUG
+        // Short-circuit if running unit tests to avoid side-effects from the app running.
+        guard !_XCTIsTesting else { return true }
+#endif
+        
+        walletLogger = OSLogger(logLevel: .debug, category: LoggerConstants.walletLogs)
+        
+        handleBackgroundTask()
+
+        return true
+    }
+
+    func application(
+        _ application: UIApplication,
+        shouldAllowExtensionPointIdentifier extensionPointIdentifier: UIApplication.ExtensionPointIdentifier
+    ) -> Bool {
+        return extensionPointIdentifier != UIApplication.ExtensionPointIdentifier.keyboard
+    }
+}
+
+// MARK: - BackgroundTasks
+
+extension AppDelegate {
+    private func handleBackgroundTask() {
+        // We require the background task to run when connected to the power and wifi
+        monitor = NWPathMonitor(requiredInterfaceType: .wifi)
+        monitor?.pathUpdateHandler = { [weak self] path in
+            if path.status == .satisfied {
+                self?.isConnectedToWifi = true
+            } else {
+                self?.isConnectedToWifi = false
+            }
+            LoggerProxy.event("BGTask isConnectedToWifi \(path.status == .satisfied)")
+        }
+        monitor?.start(queue: workerQueue)
+        
+        // set the default behavior for the NSDecimalNumber
+        NSDecimalNumber.defaultBehavior = Zatoshi.decimalHandler
+        rootStore.send(.initialization(.appDelegate(.didFinishLaunching)))
+        
+        registerTasks()
+    }
+    
+    private func registerTasks() {
+        let bcgSyncTaskResult = BGTaskScheduler.shared.register(
+            forTaskWithIdentifier: bcgTaskId,
+            using: DispatchQueue.main
+        ) { [self] task in
+            LoggerProxy.event("BGTask BGTaskScheduler.shared.register SYNC called")
+            guard let task = task as? BGProcessingTask else {
+                return
+            }
+            
+            startBackgroundTask(task)
+        }
+
+        LoggerProxy.event("BGTask SYNC registered \(bcgSyncTaskResult)")
+
+        let bcgSchedulerTaskResult = BGTaskScheduler.shared.register(
+            forTaskWithIdentifier: bcgSchedulerTaskId,
+            using: DispatchQueue.main
+        ) { [self] task in
+            LoggerProxy.event("BGTask BGTaskScheduler.shared.register SCHEDULER called")
+            guard let task = task as? BGProcessingTask else {
+                return
+            }
+
+            scheduleSchedulerBackgroundTask()
+            scheduleBackgroundTask()
+            
+            task.setTaskCompleted(success: true)
+        }
+        
+        LoggerProxy.event("BGTask SCHEDULER registered \(bcgSchedulerTaskResult)")
+    }
+    
+    private func startBackgroundTask(_ task: BGProcessingTask) {
```

Comment:
This is really clever. Scheduling tasks that always succeed will (in theory) make iOS favor the app to get more background processing time, since Apple engineers had stated in WWDC that the OS keeps a score of how well apps behave on their background time.